### PR TITLE
Minor code cleanup

### PR DESCRIPTION
**Pull request by AlexWaygood**
- Refactor `visit_ImportFrom` to reduce code duplication and improve readability.
- Minor micro-optimisation for `_has_bad_hardcoded_returns`

### CODE REVIEW

**Review by hauntsaninja — APPROVED** [review]

File: pyi.py [hunk 1]
```diff
@@ -535,15 +535,14 @@ def _has_bad_hardcoded_returns(
...
-        and return_obj_name in {"Iterable", "Iterator"}
-        and "Iterator" in bases
-    ) or (
-        method_name == "__aiter__"
-        and return_obj_name in {"AsyncIterable", "AsyncIterator"}
-        and "AsyncIterator" in bases
-    )
+    if method_name == "__iter__":
+        return return_obj_name in {"Iterable", "Iterator"} and "Iterator" in bases
+    if method_name == "__aiter__":
```
Comment (hauntsaninja) [inline pyi.py:540]:
```suggestion
    elif method_name == "__aiter__":
```

Comment (AlexWaygood) [reply pyi.py:540]: Comes to the same thing since it's straight after a `return` statement, but sure :)

Comment (hauntsaninja) [reply pyi.py]: Oh oops, not sure what I was thinking

Comment (AlexWaygood) [reply pyi.py]: Heh, well, it's in now, guess we're stuck with it until the next cleanup PR 😄

**Pull request merged**